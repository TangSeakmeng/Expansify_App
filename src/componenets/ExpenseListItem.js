import React from 'react';
import moment from 'moment';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';

import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{ description }</h3>
        </Link>
        <p>{ numeral(amount).format('$0,0.00') } - { moment(createdAt).format('MMMM Do, YYYY') }</p>
        
        {/* <button onClick={() => {
            props.dispatch(removeExpense({ id }))
        }}>Remove</button> */}
    </div>
);

export default ExpenseListItem;
