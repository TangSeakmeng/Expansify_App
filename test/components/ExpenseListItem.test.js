import React from 'react';
import { shallow } from 'enzyme';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import toJSON from 'enzyme-to-json';
import ExpenseListItem from '../../src/componenets/ExpenseListItem';
import expenses from '../fixtures/arr';

test('should expenseListItem with fixture data', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[1]} />)
    expect(wrapper).toMatchSnapshot(); 
})
