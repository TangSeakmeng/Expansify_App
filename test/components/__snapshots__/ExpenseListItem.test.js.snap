// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should expenseListItem with fixture data 1`] = `
<div>
  <[object Object]
    to="/edit/2"
  >
    <h3>
      Gas Bill
    </h3>
  </[object Object]>
  <p>
    $109,500.00
     - 
    December 28th, 1969
  </p>
</div>
`;
